examples/shader/mesh_custom_attribute.rs

```diff
@@ -88,45 +88,48 @@ fn setup(
     let mut cube_with_vertex_colors = Mesh::from(shape::Cube { size: 2.0 });
 
     // insert our custom color attribute with some nice colors!
-    cube_with_vertex_colors.set_attribute(
-        // name of the attribute
-        "Vertex_Color",
-        // the vertex attributes, represented by `VertexAttributeValues`
-        // NOTE: the attribute count has to be consistent across all attributes, otherwise bevy
-        // will panic.
-        VertexAttributeValues::from(vec![
-            // top
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-            // bottom
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-            // right
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-            // left
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-            // front
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-            // back
-            [0.79, 0.73, 0.07],
-            [0.74, 0.14, 0.29],
-            [0.08, 0.55, 0.74],
-            [0.20, 0.27, 0.29],
-        ]),
-    );
+    cube_with_vertex_colors
+        .get_mesh_data_mut()
```

Comment:
We could circumvent this "convert to mesh first then extract" pattern if we impl `From<Cube> for MeshData` and then `From<MeshData> for Mesh`. That would give us a better dataflow for cases like this and shouldn't affect ergonomics. We'd need to move PrimitiveTopology into MeshData, which I'm personally ok with. Meshes are constructed with a particular PrimitiveTopology (or at least, a particular type of topology algorithm) in mind, and its not like we support re-using MeshData across meshes anyway.